### PR TITLE
Fix: Fixed issue with pinning and unpinning shell items in Favorites

### LINKED ISSUE #11897: Bug: Cannot unpin shell folders
### Description

When you try to unpin a shell folder from the favorites, nothing happens.
Also, if you open a library (e.g. Pictures) and right click on the first path, you'll see `Pin to Favorites` even if the folder is already pinned. Clicking on it will unpin it.

### Steps To Reproduce

1. Pin some shell folders (This PC, Network, or Default Library folder)
2. Right click on a pinned item
3. Click `Unpin from favorites`
4. See that the folder is not removed

### Requirements

Files should unpin Shell folders

### Files Version

Preview 2.4.59.0

### Windows Version

Windows 11 22H2 10.0.22621.1413

### Log file
I add here the event report for the crash
[Error.txt](https://github.com/files-community/Files/files/11108570/Error.txt)

### PR DESCRIPTION
**Pull request by hecksmosis**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #11897

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [x] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open app
   2. Pin a shell folder e.g Libraries
   3. Unpin it
   4. See it doesn't crash the app and it unpins the item correctly

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App/ServicesImplementation/QuickAccessService.cs [hunk 1]
```diff
@@ -53,8 +53,10 @@ public async Task UnpinFromSidebar(string[] folderPaths)
 
 			foreach (dynamic? fi in f2.Items())
 			{
+				App.Logger.Info(fi.Path);
 				if (folderPaths.Contains((string)fi.Path)
-					|| (string.Equals(fi.Path, "::{645FF040-5081-101B-9F08-00AA002F954E}") && folderPaths.Contains(CommonPaths.RecycleBinPath)))
+					|| (string.Equals(fi.Path, "::{645FF040-5081-101B-9F08-00AA002F954E}") && folderPaths.Contains(Constants.CommonPaths.RecycleBinPath))
+					|| (string.Equals(fi.Path, "::{F02C1A0D-BE21-4350-88B0-7367FC96EF3C}") && folderPaths.Contains(Constants.CommonPaths.NetworkFolderPath)))
```
Comment (QuaintMako) [inline src/Files.App/ServicesImplementation/QuickAccessService.cs:59]: Any reason to not use the constants values in the `string.Equals`?

Comment (hecksmosis) [reply src/Files.App/ServicesImplementation/QuickAccessService.cs:59]: We can make them into constants

File: src/Files.App/UserControls/Widgets/QuickAccessWidget.xaml.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@
 using Files.App.ViewModels.Widgets;
 using Files.Backend.Services.Settings;
 using Files.Shared;
+using Files.Shared.Extensions;
```
Comment (ferrariofilippo) [inline src/Files.App/UserControls/Widgets/QuickAccessWidget.xaml.cs:11]:
```suggestion

```

File: src/Files.App/UserControls/Widgets/QuickAccessWidget.xaml.cs [hunk 3]
```diff
@@ -17,6 +18,7 @@
 using System.Collections.ObjectModel;
 using System.Collections.Specialized;
 using System.ComponentModel;
+using System.Diagnostics;
```
Comment (ferrariofilippo) [inline src/Files.App/UserControls/Widgets/QuickAccessWidget.xaml.cs:21]:
```suggestion
```

File: src/Files.App/UserControls/SidebarControl.xaml.cs [hunk 4]
```diff
@@ -190,6 +190,7 @@ private List<ContextMenuFlyoutItemViewModel> GetLocationItemMenuItems(INavigatio
 
 			var isDriveItem = item is DriveItem;
 			var isDriveItemPinned = isDriveItem && ((DriveItem)item).IsPinned;
+			var isShellFolder = item is LocationItem shellItem && shellItem.Path.StartsWith("\\\\SHELL\\");
```
Comment (ferrariofilippo) [inline src/Files.App/UserControls/SidebarControl.xaml.cs:193]:
```suggestion
			var isShellFolder = item is LocationItem shellItem && ShellStorageFolder.IsShellPath(shellItem.Path);
```

File: src/Files.App/ServicesImplementation/JumpListService.cs [hunk 5]
```diff
@@ -110,6 +110,10 @@ private void AddFolder(string path, string group, JumpList instance)
 			if (instance is not null)
 			{
 				string displayName = null;
```
Comment (ferrariofilippo) [inline src/Files.App/ServicesImplementation/JumpListService.cs:112]:
```suggestion
				string? displayName = null;
```

File: src/Files.App/ServicesImplementation/JumpListService.cs [hunk 6]
```diff
@@ -110,6 +110,10 @@ private void AddFolder(string path, string group, JumpList instance)
...
+				
+				if (path.StartsWith("\\\\SHELL", StringComparison.OrdinalIgnoreCase))
+					displayName = "ThisPC".GetLocalizedResource();
+				
 				if (path.EndsWith("\\"))
```
Comment (ferrariofilippo) [inline src/Files.App/ServicesImplementation/JumpListService.cs:117]:
```suggestion
				if (path.EndsWith('\\'))
```

**Review by ferrariofilippo — CHANGES_REQUESTED** [review]
I left some comments on code style and on how to fix build errors.
BTW I can't unpin `This PC`

File: src/Files.App/ServicesImplementation/QuickAccessService.cs [hunk 7]
```diff
@@ -53,8 +56,22 @@ public async Task UnpinFromSidebar(string[] folderPaths)
...
 			foreach (dynamic? fi in f2.Items())
 			{
-				if (folderPaths.Contains((string)fi.Path)
-					|| (string.Equals(fi.Path, "::{645FF040-5081-101B-9F08-00AA002F954E}") && folderPaths.Contains(CommonPaths.RecycleBinPath)))
+				if (ShellStorageFolder.IsShellPath((string)fi.Path))
+				{
+					var folder = await ShellStorageFolder.FromPathAsync((string)fi.Path);
+					var path = folder.Path;
+
+					if (folderPaths.Contains(path) || (path.StartsWith(@"\\SHELL\") && folderPaths.Any(x => x.StartsWith(@"\\SHELL\")))) // Fix for the Linux header
```
Comment (QuaintMako) [inline src/Files.App/ServicesImplementation/QuickAccessService.cs:64]: Condition on two lines?

**Review by QuaintMako — APPROVED** [review]
LGTM and works as expected on my machine. I can pin and unpin shell items.

**Review by ferrariofilippo — APPROVED** [review]
LGTM and works now

**Pull request merged**